### ISSUE TITLE
CSD05 Ballot Comment - Numbers

### ISSUE THREAD

**Issue by sparrell** [oasis-tcs/openc2-oc2ls#98]
Oc2ls section 3.1.2 states 
`“For JSON format all idiomatic types shown here are converted (if necessary) to String representation before serialization."`
 For example UUID's are shown as "71be3c32-188f-476d-9b20-35cb4eb60e52". JSON allows for numbers - why are we converting to strings? I recommend we use numbers for machine2machine JSON when the underlying type is a number.

**Comment by varnerac** [comment]
I've never seen UUIDs in JSON as numbers. To process a UUID as in integer, you need support for 128-bit integers (`double double`) or arbitrary length integers, which are a pain. Additionally, the string representation of a UUID is idiomatic, explicitly defined in [Part 3 of RFC 412](https://tools.ietf.org/html/rfc4122#section-3). I've only seen strings in JSON and sometimes `bytes` in Google Protocol Bufffers.

**Comment by sparrell** [comment]
OK so maybe UUID was bad example for using a number. Should all numbers be converted to strings? I though JSON allowed numbers without the quotes. { "my_age" : 62 ) instead of {"my_age" : "62"}.

**Comment by varnerac** [comment]
I think that JSON serialization should have the option for whatever makes sense, not just all strings. For example, if we have to serialize a "count" in JSON, it makes no sense for the count to be `"3"` instead of `3`. If something is a number, it should be a number. If it should be a string, it should be a string. I don't think it makes sense to mandate that all JSON values are serialized as strings.

**Comment by dmg2** [comment]
I think most of us agree with varnerac's comment above, I do not think anything in the language precludes us from doing this. additionally the way we have written our property tables and the way we validate our JSON scheme today is congruent with this thought process. We should be able to close out this comment.

**Comment by davaya** [comment]
DateTime is an example - we have defined it as a number, the number of milliseconds since the epoch (Jan 1 1970), and in JSON we serialize it as a number.  Status Code is another, we chose to transmit 200 instead of the string "OK".    But other values such as IPv4 Addr, MAC Addr, Hash, etc, we've chosen to serialize as idiomatic strings (dotted quad for IP addresses and hex strings for Hashes.  We currently serialize MAC Addr as a hex string too (A04392162FBC) instead of the usual colon-hex string (A0:43:92:16:2F:BC), but if we changed our minds, we could define a colon-hex format and use it.

We should be able to close the comment - the property tables allow numbers to be serialized as JSON numbers if desired, or as various types of strings if desired, using the `/format` notation.
